### PR TITLE
Event mididleware system

### PR DESCRIPTION
**Pull request by Lunarmagpie**
guild_typing_event is just an example. We would use `default` in the actual bot probably.

### CODE REVIEW

File: bot/core/event_middleware.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
+from __future__ import annotations
```
Comment (CircuitSacul) [inline bot/core/event_middleware.py:1]: This file can go under core/

**Review by CircuitSacul — CHANGES_REQUESTED** [review]
After the one comment, LGTM

File: bot/core/event_middleware.py (hunk 1 above)
Comment (Lunarmagpie) [reply bot/core/event_middleware.py:1]: it is :skull:

Comment (CircuitSacul) [reply bot/core/event_middleware.py:1]: Oh

**Review by CircuitSacul — APPROVED** [review]

**Pull request merged**